src/main/webapp/js/new-post-loader.js

```diff
@@ -1,3 +1,174 @@
+  setupClickListener('changetype-geocode', ['geocode']);
+
+  document.getElementById('use-strict-bounds')
+      .addEventListener('click', function() {
+        console.log('Checkbox clicked! New state=' + this.checked);
+        autocomplete.setOptions({strictBounds: this.checked});
+      });
+}
+
+
+function setCenterToCurrentLoc(map){
+  var infoWindow = new google.maps.InfoWindow();
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+      var pos = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude
+      };
+
+      infoWindow.setPosition(pos);
+      infoWindow.setContent('Current Location Detected');
+      infoWindow.open(map);
+      map.setCenter(pos);
+      currentPos = pos;
+    }, function() {
+      handleLocationError(map, true, infoWindow, map.getCenter());
+    });
+  } else {
+    // Browser doesn't support Geolocation
+    handleLocationError(map, false, infoWindow, map.getCenter());
+  }
+}
+
+function handleLocationError(map, browserHasGeolocation, infoWindow, pos) {
+  infoWindow.setPosition(pos);
+  infoWindow.setContent(browserHasGeolocation ?
+                        'Error: The Geolocation service failed.' :
+                        'Error: Your browser doesn\'t support geolocation.');
+  infoWindow.open(map);
+}
+
+function addLocationInfoToForm(place){
+  var latInput = document.createElement("input");
+  latInput.setAttribute("type", "hidden");
+  latInput.setAttribute("name", "lat");
+  latInput.setAttribute("value", place.geometry.location.lat());
+
+  var lngInput = document.createElement("input");
+  lngInput.setAttribute("type", "hidden");
+  lngInput.setAttribute("name", "lng");
+  lngInput.setAttribute("value", place.geometry.location.lng());
+
+  var nameInput = document.createElement("input");
+  nameInput.setAttribute("type", "hidden");
+  nameInput.setAttribute("name", "location_name");
+  nameInput.setAttribute("value", place.name);
+
+  var idInput = document.createElement("input");
+  idInput.setAttribute("type", "hidden");
+  idInput.setAttribute("name", "location_id");
+  idInput.setAttribute("value", place.id);
+
+  document.getElementById("message-form").appendChild(latInput);
+  document.getElementById("message-form").appendChild(lngInput);
+  document.getElementById("message-form").appendChild(nameInput);
+  document.getElementById("message-form").appendChild(idInput);
+}
+
+function replaceLocationInfo(place){
+  document.getElementsByName("lat")[0].value = place.geometry.location.lat();
+  document.getElementsByName("lng")[0].value = place.geometry.location.lng();
+  document.getElementsByName("location_name")[0].value = place.name;
+  document.getElementsByName("location_id")[0].value = place.id;
+}
+
+function showLocationChosen(name){
+  var message = document.getElementById('location-message');
+  message.innerHTML = message.innerHTML + name;
```

Comment:
Small bug, if you select multiple locations it appends them to each other. Maybe just set it without appending, something like `message.innerHTML = "Location Chosen: " + name`